package.json

```diff
@@ -3,25 +3,29 @@
   "version": "3.0.2",
   "description": "Utilities for dealing with cross-domain windows.",
   "main": "dist/cross-domain-utils.js",
-  "module": "dist/module/index.js",
+  "module": "dist/index.js",
+  "types": "dist/esm/index.d.ts",
+  "sideEffects": false,
   "scripts": {
-    "build": "npm run test && npm run babel && npm run webpack",
+    "build": "npm run test && npm run babel && npm run webpack && npm run build:types",
+    "build:flow": "find ./dist -type f -not -path './node_modules/*' -name '*.d.ts' -exec sh -c 'flowgen --add-flow-header $1 -o ${1%.*.*}.js.flow' _ '{}' \\;",
+    "build:tsc": "tsc src/* --outDir ./dist/esm --declaration --emitDeclarationOnly --strict",
+    "build:types": "npm run build:tsc && npm run build:flow",
     "webpack": "cross-env NODE_ENV=production babel-node --plugins=transform-es2015-modules-commonjs ./node_modules/.bin/webpack --progress --output-path dist",
-    "babel": "cross-env NODE_ENV=production babel src/ --out-dir dist/module",
-    "karma": "cross-env NODE_ENV=test babel-node --plugins=transform-es2015-modules-commonjs ./node_modules/.bin/karma start",
+    "babel": "cross-env NODE_ENV=production babel src/ --out-dir ./dist/esm/ --extensions .ts,.tsx",
+    "tsc": "tsc",
```

Comment:
Do we want to follow the convention of having a `typecheck` script here, like we have in other repositories?